### ISSUE TITLE
riscv/fastcpu: UMul/IMul versus their description

### ISSUE THREAD

**Issue by neuschaefer** [s-macke/jor1k#104]
In `js/worker/riscv/fastcpu.js`, there are the two functions `UMul` and `IMul`. `UMul` is documented as a signed\*signed multiplication and `IMul` as unsigned\*unsigned multiplication. However, their names suggest that it's in fact the other way around (`UMul` sounds like unsigned multiplication).

Which function does what?

**Comment by andrakis** [comment]
If I understand you correctly, you want to know whether jor1k is doing what the (incorrect?) documentation says, or the "correct" behavior that occurs in a real device.

According to [the imul section](https://github.com/s-macke/jor1k/blob/master/js/worker/or1k/fastcpu.js#L1395), it's been implemented as per existing real devices, not what the documentation says.

**Comment by neuschaefer** [comment]
I think a was a little tired when I wrote the above :)

I'm talking about [fastcpu.js](https://github.com/s-macke/jor1k/blob/master/js/worker/riscv/fastcpu.js#L853) in the (experimental?) RISC-V code. And by documentation I mean the comment in the third line of these functions.

By reading some more of the code, I found out that the `mulh` instruction (signed\*signed) uses `UMul` and the `mulhu` instruction (unsigned\*unsigned) uses `IMul`. So the comments are correct, but the names are wrong (IMHO, at least).
